docs/guides/websockets/chat/javascript.md

```diff
@@ -0,0 +1,292 @@
+
+Create a new Role called `Users`, and create a new user with this role. Make note of the password you set.
+
+Within the `Users` role, give Create and Read access to the `Messages` collection, and Read access to the `Directus Users` system collection. 
+
+### Create a HTML Boilerplate
+
+Create an `index.html` file and open it in your code editor:
+
+```html
+<!DOCTYPE html>
+<html>
+<body>
+  <form id="login">
+    <label for="email">Email</label>
+    <input type="email" id="email">
+    <label for="password">Password</label>
+    <input type="password" id="password">
+    <input type="submit">
+  </form>
+
+  <ol></ol>
+
+  <form id="new">
+    <label for="message">Message</label>
+    <input type="text" id="text">
+    <input type="submit">
+  </form>
+
+  <script>
+    
+  </script>
+</body>
+</html>
+```
+
+The first form will handle user login and the second will handle new message submissions. The empty `<ol>` will be populated with messages. 
+
+Inside of the `<script>`, create a `url` variable being sure to replace `your-directus-url` with your project’s URL:
+
+```js
+const url = 'wss://your-directus-url/websocket'
+let socket
+```
+
+The `socket` variable will later contain a WebSocket instance.
+
+Finally, create event listeners which are triggered on the form submissions:
+
+```js
+document.querySelector('#login').onsubmit = function(event) {
+  event.preventDefault()
+}
+
+document.querySelector('#new').onsubmit = function(event) {
+  event.preventDefault()
+}
+```
+
+## Establish WebSocket Connection
+
+Within the `#login` form submit event handler, extract the `email` and `password` values from the form:
+
+```js
+const email = event.target.elements.email.value
+const password = event.target.elements.password.value
+```
+
+Create a new WebSocket, which will immediately attempt connection:
+
+```js
+socket = new WebSocket(url)
+```
+
+On connection, you must [send an authentication message before the timeout](/guides/websockets/authentication). Add an event handler for the socket’s `open` event:
+
+```js
+socket.onopen = function() {
+  socket.send(JSON.stringify({ 
+    type: 'auth', 
+    email, 
+    password 
+  }))
+}
+```
+
+## Subscribe To Messages
+
+In a WebSocket connection, all data sent from the server will trigger the socket’s `message` event. Underneath the `socket.onopen` function, add the following:
+
+```js
+socket.onmessage = function(message) {
+  receiveMessage(message)
+}
+```
+
+At the bottom of your `<script>`, create the `receiveMessage` function:
+
+```js
+function receiveMessage(message) {
+  const data = JSON.parse(message.data)
+}
+```
+
+As soon as you have successfully authenticated, a message will be sent. When this happens, subscribe to updates on the `Messages` collection. Add this inside of the `receiveMessage` function:
+
+```js
+if(data.type == 'auth' && data.status == 'ok') {
```

Comment:
```suggestion
if (data.type == 'auth' && data.status == 'ok') {
```

Reply:
Ah we enter the world of opinionated suggestions 😉 I never put a space there.

Reply:
All opinions aside, we should really be following our own codebase style rules in these examples.

Reply:
Hahaha in that case yes, this is very much my opinion for readability 🙈